src/components/NcRelatedResourcesPanel/NcTeamResources.vue

```diff
@@ -0,0 +1,293 @@
+import NcActionLink from '../NcActionLink/NcActionLink.vue'
+import NcIconSvgWrapper from '../NcIconSvgWrapper/NcIconSvgWrapper.vue'
+
+import { t } from '../../l10n.js'
+
+export default {
+	name: 'NcTeamResources',
+
+	components: {
+		AccountGroup,
+		ChevronDown,
+		ChevronUp,
+		OpenInNew,
+		NcButton,
+		NcActions,
+		NcActionLink,
+		NcIconSvgWrapper,
+	},
+
+	props: {
+		providerId: {
+			type: String,
+			default: null,
+		},
+		itemId: {
+			type: [String, Number],
+			default: null,
+		},
+		resourceType: {
+			type: String,
+			default: '',
+		},
+		/**
+		 * Set the maximum number of resources to load
+		 */
+		limit: {
+			type: Number,
+			default: 0,
+		},
+		/**
+		 * Only used by the files sidebar
+		 *
+		 * File info is passed when registered with `OCA.Sharing.ShareTabSections.registerSection()`
+		 */
+		fileInfo: {
+			type: Object,
+			default: null,
+		},
+		/**
+		 * Make the header name dynamic
+		 */
+		header: {
+			type: String,
+			default: t('Related resources'),
+		},
+		headerOther: {
+			type: String,
+			default: t('Other resources'),
+		},
+		description: {
+			type: String,
+			default: t('Anything shared with the same group of people will show up here'),
+		},
+		/**
+		 * If this element is used on a primary element set to true for primary styling.
+		 */
+		primary: {
+			type: Boolean,
+			default: false,
+		},
+	},
+
+	data() {
+		return {
+			appEnabled: OC?.appswebroots?.circles !== undefined && (OC.config.version.split('.')[0] ?? 0) >= 29,
+			loading: false,
+			error: null,
+			teamResources: null,
+			teamOpen: null,
+		}
+	},
+
+	computed: {
+		isVisible() {
+			return !this.loading && this.error === null && this.teamResources?.length > 0
+		},
+		teamProviders() {
+			return (teamId) => {
+				const team = this.teamResources.find(t => t.teamId === teamId)
+				return team.resources?.reduce((acc, resource) => {
+					if (resource.provider.id === this.providerId && resource.id === String(this.itemId)) {
+						return acc
+					}
+
+					if (!acc[resource.provider.id]) {
+						acc[resource.provider.id] = resource.provider
+						acc[resource.provider.id].resources = []
+					}
+
+					if (resource.provider.id === this.providerId && resource.id === String(this.itemId)) {
+						return acc
+					}
+
+					acc[resource.provider.id].resources.push(resource)
+					return acc
+				}, {})
+			}
+		},
+		open() {
+			return (teamId) => {
+				return this.teamOpen === teamId
+			}
+		},
+	},
+
+	watch: {
+		providerId() {
+			this.fetchTeamResources()
+		},
+		itemId() {
+			this.fetchTeamResources()
+		},
+	},
+
+	created() {
+		this.fetchTeamResources()
+	},
+
+	methods: {
+		t,
+		async fetchTeamResources() {
+			try {
+				const response = await axios.get(generateOcsUrl(`/teams/resources/${this.providerId}/${this.itemId}`))
+				this.error = null
+				this.teamResources = response.data.ocs.data.teams
+				this.teamOpen = this.teamResources[0]?.teamId
+			} catch (e) {
+				this.error = e
```

Comment:
Maybe rather reset `this.teamResources = null` here and get rid of `this.error`? It's not displayed anywhere in UI anyway.